src/api/app/models/bs_request_action_maintenance_incident.rb

```diff
@@ -103,6 +103,36 @@ def expand_targets(ignore_build_state, ignore_delegate)
     super(ignore_build_state, ignore_delegate)
   end
 
+  def modify_sources(force_branching)
+    # is branch enforcement a policy?
+    maintenance_project = Project.find_by_name(target_project)
+    return if force_branching.nil? && Attrib.find_by_container_and_fullname(maintenance_project, "OBS:EnforceIncidentRequestStaging").nil?
+
+    title = 'Enforce branch project for maintenance incident request'
+    description = ''
+    stage_project = nil
+    Project.transaction do
+      # target projects may differ between the actions, we just need a uniq temporary request project
+      # we use select instead of where() because this is not yet stored in DB.
+      first_incident_action = bs_request.bs_request_actions.select{ |v| v.type == 'maintenance_incident' }.first
+      stage_project = Project.create(name: User.session!.branch_project_name(first_incident_action.target_releaseproject),
+                                     title: title, description: description)
+      stage_project.relationships.build(user: User.session!, role: Role.find_by_title!('maintainer'))
+      stage_project.store
+      # autocleanup attribute in case request gets not accepted?
+    end
+
+    # create package
+    pkg = _merge_pkg_into_maintenance_incident(stage_project)
+    # create channels
+    pkg.add_channels
+    # adapt request action
+    self.source_package = pkg.name
+    self.source_project = stage_project.name
+    # create patchinfo
+    Patchinfo.new.create_patchinfo_from_request(stage_project, bs_request)
+  end
```

Comment:
this code is doing too many things at the same time. Could you break it in smaller methods?